src/nats.h

```diff
@@ -1249,115 +1370,9 @@ typedef struct jsOptions
-
-                /// @brief When using the automatic Fetch flow control (default
-                /// NextHandler), initiate the next fetch request (this many
-                /// messages) prior to the fulfillment of the current request.
-                ///
-                /// @note KeepAhead can not be used in conjunction with MaxBytes
-                /// or NoWait.
-                int                     KeepAhead;
-
-                /// @brief If set, switches to manual fetch flow control.
-                ///
-                /// If provided, this function gets called before each message
-                /// is deliverered to msgCB, and overrides the default algorithm
-                /// for sending Next fetch requests.
-                jsFetchNextHandler      NextHandler;
-                void                    *NextHandlerClosure;
-
-        } PullSubscribeAsync; ///< extra options for #js_PullSubscribeAsync
-
-
-        /**
-         * Advanced stream options
-         *
-         * * `Purge` for advanced purge options.
-         * * `Info` for advanced information retrieval options.
-         */
-        struct jsOptionsStream
-        {
-                /**
-                 * Advanced stream purge options
-                 *
-                 * * `Subject` will filter the purge request to only messages that match the subject, which can have wildcards.<br>
-                 * * `Sequence` will purge up to but not including this sequence and can be combined with subject filtering.<br>
-                 * * `Keep` will specify how many messages to keep and can be combined with subject filtering.<br>
-                 *
-                 * \note `Sequence` and `Keep` are mutually exclusive, so both can not be set at the same time.
-                 */
-                struct jsOptionsStreamPurge
-                {
-                        const char      *Subject;       ///< This is the subject to match against messages for the purge command.
-                        uint64_t        Sequence;       ///< Purge up to but not including sequence.
-                        uint64_t        Keep;           ///< Number of messages to keep.
-
-                } Purge;                                ///< Optional stream purge options.
-
-                /**
-                 * Advance stream information retrieval options
-                 */
-                struct jsOptionsStreamInfo
-                {
-                        bool            DeletedDetails;         ///< Get the list of deleted message sequences.
-                        const char      *SubjectsFilter;        ///< Get the list of subjects in this stream.
-
-                } Info;                                 ///< Optional stream information retrieval options.
-
-        } Stream;                                       ///< Optional stream options.
+        jsOptionsPublishAsync PublishAsync;                     ///< extra options for #js_PublishAsync
```

Comment:
nit: If you could align `PublishAsync;`, `PullSubscribeAsync;` and `Stream` fields, that would be nice.